### ISSUE TITLE
Returning aria-label prop from Button component.

### ISSUE THREAD

**Issue by MohammedRAihanAhmed** [cerner/terra-consumer#108]
# Issue Description #

The "Button" component in

* terra-consumer-nav/src/components/nav-burger-button/NavBurgerButton.jsx
* terra-consumer-nav/src/components/Nav.jsx

does not have an "aria-label" attribute associated with it. This is required to achieve Level A, WCAG Compliance.

This is the error the HTML code sniffer spits out when run on DEx for the above mentioned Button
* This button element does not have a name available to an accessibility API.
  Valid names are: title attribute, element content, aria-label attribute, aria-labelledby attribute.

The following screenshot of the HTML code sniffer which was run on DEx shows the error and its description.

<img width="304" alt="screen shot 2018-01-15 at 2 08 05 pm" src="https://user-images.githubusercontent.com/7814781/34996726-1cacb764-faa0-11e7-8151-593b639271c0.png">

<img width="304" alt="screen shot 2018-01-15 at 2 08 23 pm" src="https://user-images.githubusercontent.com/7814781/34996748-2ceff3fc-faa0-11e7-8db6-530371204d85.png">

1. A proposed fix would be to add a custom prop to the Button component as "aria-label='some-name'".
2. A different approach would be to make the Button component(which  is consumed from 'terra-button'), to return aria-label if the following conditions are met.
    * prop 'isIconOnly' is true.
    * 'text' value is provided. 
  
The following line of code from 'terra-button' explains that 'aria-label' will be set to 'text' if 'isIconOnly' is passed as true - https://github.com/cerner/terra-core/blob/master/packages/terra-button/src/Button.jsx#L216.

To run HTMLCodesniffer, get the bookmarklet [here](http://squizlabs.github.io/HTML_CodeSniffer/)

# Issue Type #

- [ ] New Feature
- [ ] Enhancement
- [ ] Bug
- [x] Other

## Expected Behavior ##

The Button component should have an "aria-label" attribute assigned to it.

<img width="913" alt="screen shot 2018-01-15 at 2 33 39 pm" src="https://user-images.githubusercontent.com/7814781/34996786-4410ee10-faa0-11e7-98b5-f4774fbe2181.png">

## Current Behavior ##

The Button Component does not have "aria-label" attribute that leads to the error which is caught by
the HTMLCodesniffer.

<img width="763" alt="screen shot 2018-01-15 at 2 38 27 pm" src="https://user-images.githubusercontent.com/7814781/34996804-548bd728-faa0-11e7-9fb4-0a73bf700497.png">

## Steps to Reproduce ##
1. Get the [HTMLCodesniffer](http://squizlabs.github.io/HTML_CodeSniffer/) bookmarklet.
2. Run the [terra-consumer-site](https://github.com/cerner/terra-consumer)
3. Navigate to 'Tests > Nav Tests > Burger Button'
4. Click on the HTMLCodesniffer bookmark to re produce the error.

**Comment by amf9t2** [comment]
Terra-button with isIconOnly prop is not released. It is expected to release soon. Hoping that will fix the issue, only change required in terra-consumer will be setting "IsIconOnly" prop.
